### PR TITLE
2.x.x

### PR DESCRIPTION
**Pull request by tomkis1**
## Overview 

- getting rid of generators
- introducing composable sagas for LLTs and side effects
- View HOC
- Merging `exactMatcher` into default `matcher` implementation

## Todo

- [x] implementation
- [x] examples
- [x] documentation
- [x] jsdocs
- [ ] unit tests
- [x] devtools / replay / undo / redo
- [x] redux-saga-rxjs ?
- [x] get rid of redux-saga peerDep

### CODE REVIEW

**Comment by tomsdev** [comment]
Hey, I really like the changes in here. I was wondering if the implementation here will be flexible enough to allow to use redux-saga-rxjs instead of redux-saga? I think that'd be great as not everyone can or want to use generators.

**Comment by tomkis1** [comment]
@tomsdev Absolutely! will need to figure this out though

File: README.md [hunk 1]
```diff
@@ -29,41 +29,37 @@ See [why the Elm Architecture matters](http://salsita.github.io/redux-elm/).
...
 
-export default ({ model, dispatch }) => (
+export default view(({ model, dispatch }) => (
   <div>
     <button onClick={() => dispatch({ type: 'Decrement' })}>-</button>
     <div>{model}</div>
     <button onClick={() => dispatch({ type: 'Increment' })}>+</button>
   </div>
-);
+));
```
Comment (matthewgertner) [inline README.md]: Nothing about sagas in the README?

File: examples/gif-viewers-pair/src/main.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+require('isomorphic-fetch');
```
Comment (matthewgertner) [inline examples/gif-viewers-pair/src/main.js]: Can't you say `import 'isomorphic-fetch';`?

File: src/Updater.js [hunk 3]
```diff
@@ -1,122 +1,138 @@
...
  */
+const instantiateSaga = (
+  saga,
+  stateRepository,
+  subscribersRepository,
+  actionPrefix,
+  dispatch
+) => runSaga(saga(), {
+  subscribe: cb => {
+    if (!subscribersRepository[actionPrefix]) {
```
Comment (matthewgertner) [inline src/Updater.js]: Would be cleaner IMO to say something like:

```
const actionPrefixSubscribers = subscribers[actionPrefix] || [];
actionPrefixSubscribers.push(cb);
...
```

File: src/Updater.js [hunk 4]
```diff
@@ -1,122 +1,138 @@
...
     const matcher = matcherImpl ? matcherImpl(pattern) : this.defaultMatcherImpl(pattern);
     this.matchers.push({ matcher, updater });
 
     return this;
   }
 
   /**
-   * Converts Updater to Redux compatible Reducer.
-   * You always need to call this method before exporting the Updater.
+   * Converts Updater to Redux comaptible plain old function
```
Comment (matthewgertner) [inline src/Updater.js]: "compatible"

**Pull request merged**